discovery_engine/lib/src/ffi/types/document/document_vec.dart

```diff
@@ -0,0 +1,70 @@
+// Copyright 2022 Xayn AG
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, version 3.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+import 'dart:ffi' show Pointer;
+
+import 'package:xayn_discovery_engine/src/ffi/genesis.ffigen.dart'
+    show RustDocument, RustDocumentVec;
+import 'package:xayn_discovery_engine/src/ffi/load_lib.dart' show ffi;
+import 'package:xayn_discovery_engine/src/ffi/types/document/document.dart';
+
+extension DocumentSliceFfi on List<Document> {
+  /// Allocates a slice of documents containing all document of this list.
+  ///
+  /// We expect the length of this list to be passed in.
+  ///
+  //Note: The len arguments makes sure the len you use with the slice and
+  //      the len you create it with are the same, alternatively returning
+  //      a `Pair` or custom type could be done.
+  Pointer<RustDocument> createSlice(final int len) {
```

Comment:
I don't understand why we need this. Both we and the caller have len.

Reply:
The idea is to avoid some situations where the length could have changed.

But tbh. it's me being overly cautious and not worth it at all. I will remove it.